apps/android/LiveVideoApp/app/src/main/java/com/twilio/livevideo/app/manager/room/RoomManager.kt

```diff
@@ -0,0 +1,165 @@
+package com.twilio.livevideo.app.manager.room
+
+import android.content.Context
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.Observer
+import com.twilio.livevideo.app.custom.BaseLifeCycleComponent
+import com.twilio.livevideo.app.repository.model.ErrorResponse
+import com.twilio.video.BandwidthProfileMode
+import com.twilio.video.OpusCodec
+import com.twilio.video.RemoteParticipant
+import com.twilio.video.Room
+import com.twilio.video.TrackPriority
+import com.twilio.video.TwilioException
+import com.twilio.video.Vp8Codec
+import com.twilio.video.ktx.ConnectOptionsBuilder
+import com.twilio.video.ktx.Video
+import com.twilio.video.ktx.createBandwidthProfileOptions
+import timber.log.Timber
+import javax.inject.Inject
+
+class RoomManager @Inject constructor(
+    private val context: Context?,
+    private val localParticipantWrapper: LocalParticipantWrapper
+) : BaseLifeCycleComponent(),
+    Room.Listener {
+
+    private val _onStateEvent: MutableLiveData<RoomViewEvent?> =
+        MutableLiveData<RoomViewEvent?>(null)
+    val onStateEvent: LiveData<RoomViewEvent?>
+        get() {
+            val event = _onStateEvent
+            _onStateEvent.value = null
+            return event
+        }
+
+    private val participantsObserver: Observer<in RoomViewEvent?> = Observer {
+        it?.apply {
+            this@RoomManager._onStateEvent.value = this
+        }
+    }
+
+    private var lifecycleOwner: LifecycleOwner? = null
+    private var room: Room? = null
+    private var participants: MutableList<ParticipantStream>? = null
+
+    fun connect(
+        lifecycleOwner: LifecycleOwner,
+        roomName: String,
+        accessToken: String,
+        isHost: Boolean = false
+    ) {
+        this.lifecycleOwner = lifecycleOwner
+        init(lifecycleOwner.lifecycle)
+        localParticipantWrapper.init(lifecycleOwner.lifecycle)
+        localParticipantWrapper.isHost = isHost
+        room = context?.let {
+            val connectOptions: ConnectOptionsBuilder = {
+                this.roomName(roomName)
+                this.audioTracks(listOf(localParticipantWrapper.localAudioTrack))
+                this.videoTracks(listOf(localParticipantWrapper.videoTrack))
+                this.enableDominantSpeaker(true)
+                this.bandwidthProfile(createBandwidthProfileOptions {
+                    this.mode(BandwidthProfileMode.GRID)
+                    this.dominantSpeakerPriority(TrackPriority.HIGH)
+                })
+                preferVideoCodecs(listOf(Vp8Codec()))
```

Comment:
I'm thinking we want VP8 simulcast. Is that not what the Android video app uses by default?

Reply:
Yes, according to `AndroidVideoRoomApp` this is the default video codec(without simulcast). However, if we want to use `VP8 simulcast`, when I just need to pass true as parameter. Something like: `Vp8Codec(true)`. Let me know.

Reply:
Done!